core/src/main/java/org/mule/runtime/core/privileged/event/PrivilegedEvent.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.privileged.event;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.api.metadata.DataType;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.connector.ReplyToHandler;
+import org.mule.runtime.core.api.event.BaseEvent;
+import org.mule.runtime.core.api.event.BaseEventContext;
+import org.mule.runtime.core.api.event.MuleSession;
+import org.mule.runtime.core.api.source.MessageSource;
+import org.mule.runtime.core.api.transformer.MessageTransformerException;
+
+import java.nio.charset.Charset;
+
+/**
+ * Allows access to the privileged behavior of the {@link BaseEvent}.
+ *
+ * @since 4.0
+ */
+public interface PrivilegedEvent extends BaseEvent {
+
+  class CurrentEventHolder {
+
+    private static final ThreadLocal<BaseEvent> currentEvent = new ThreadLocal<>();
+  }
+
+  /**
+   * @return the context applicable to all events created from the same root {@link BaseEvent} from a {@link MessageSource}.
+   */
+  @Override
+  BaseEventContext getContext();
+
+  /**
+   * @return the correlation id to use for this event.
+   * @deprecated TODO MULE-10706 Mule 4: remove this
+   */
+  @Deprecated
+  String getLegacyCorrelationId();
+
+  /**
+   * Retrieves the service session for the current event
+   *
+   * @return the service session for the event
+   * @deprecated Transport infrastructure is deprecated.
+   */
+  @Deprecated
+  MuleSession getSession();
+
+  /**
+   * Return the replyToHandler (if any) that will be used to perform async reply
+   *
+   * @deprecated TODO MULE-10739 Move ReplyToHandler to compatibility module.
+   */
+  @Deprecated
+  ReplyToHandler getReplyToHandler();
+
+  /**
+   * Return the destination (if any) that will be passed to the reply-to handler.
+   *
+   * @deprecated TODO MULE-10739 Move ReplyToHandler to compatibility module.
+   */
+  @Deprecated
+  Object getReplyToDestination();
+
+  /**
+   * Return the event associated with the currently executing thread.
+   *
+   * @return event for currently executing thread.
+   */
+  static BaseEvent getCurrentEvent() {
+    return CurrentEventHolder.currentEvent.get();
```

Comment:
static import

Reply:
CurrentEventHolder is an inner class, its members cannot be statically imported